app/components/channel_drawer/channels_list/switch_teams/switch_teams.js

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+    Text,
+    TouchableHighlight,
+    View
+} from 'react-native';
+import AwesomeIcon from 'react-native-vector-icons/FontAwesome';
+
+import Badge from 'app/components/badge';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+export default class SwitchTeams extends React.PureComponent {
+    static propTypes = {
+        currentTeam: PropTypes.object.isRequired,
+        searching: PropTypes.bool.isRequired,
+        showTeams: PropTypes.func.isRequired,
+        teamMembers: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            badgeCount: this.getBadgeCount(props)
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.currentTeam !== this.props.currentTeam || nextProps.teamMembers !== this.props.teamMembers) {
+            this.setState({
+                badgeCount: this.getBadgeCount(nextProps)
+            });
+        }
+    }
+
+    getBadgeCount = (props) => {
+        const {
+            currentTeam,
+            teamMembers
+        } = props;
+
+        let mentionCount = 0;
+        let messageCount = 0;
+        Object.values(teamMembers).forEach((m) => {
+            if (m.team_id !== currentTeam.id) {
+                mentionCount = mentionCount + (m.mention_count || 0);
+                messageCount = messageCount + (m.msg_count || 0);
+            }
+        });
+
+        let badgeCount;
+        if (mentionCount) {
+            badgeCount = mentionCount;
+        } else if (messageCount) {
+            badgeCount = -1;
+        } else {
+            badgeCount = 0;
+        }
+
+        return badgeCount;
+    };
+
+    showTeams = wrapWithPreventDoubleTap(() => {
+        this.props.showTeams();
+    });
+
+    render() {
+        const {
+            currentTeam,
+            searching,
+            teamMembers,
+            theme
+        } = this.props;
+
+        const {
+            badgeCount
+        } = this.state;
+
+        if (searching || teamMembers.length < 2) {
```

Comment:
with the latests changes for the drawer, the teams tab is always present as you can see here https://github.com/mattermost/mattermost-mobile/blob/master/app/components/channel_drawer/channel_drawer.js#L310 and then the swiper decides to show the pagination and to enable the scroll if there's a need for that https://github.com/mattermost/mattermost-mobile/blob/master/app/components/channel_drawer/drawer_swipper/drawer_swiper.js#L85

If we don't do it that way the whole swiper goes crazy

Reply:
This is just for the button in the top-right that switches teams. I didn't change how the pages are rendered in the swiper